docs/src/upgrades.md

```diff
@@ -14,6 +14,88 @@ changed by default.
 
 Use `bootc upgrade --apply` to auto-apply if there are queued changes.
 
+### Staged updates with `--download-only`
+
+The `--download-only` flag allows you to prepare updates without automatically applying
+them on the next reboot:
+
+```shell
+bootc upgrade --download-only
+```
+
+This will pull the new container image from the registry and create a staged deployment
+that is **locked** from automatic application. The deployment will not be applied on
+shutdown or reboot until it is explicitly unlocked.
+
+#### Checking lock status
+
+To see whether a staged deployment is locked, use:
+
+```shell
+bootc status --verbose
+```
+
+In the output, you'll see `Locked: yes` for locked deployments or `Locked: no` for unlocked
+deployments. The lock status is only shown in verbose mode.
+
+#### Applying locked updates
+
+There are two ways to apply a locked staged update:
+
+**Option 1: Apply immediately with reboot**
+
+```shell
+bootc upgrade --apply
+```
+
+This will unlock the staged deployment (if locked) and immediately reboot into it.
+
+**Option 2: Unlock for automatic application**
+
+```shell
+bootc upgrade
+```
+
+Running `bootc upgrade` without flags on an already-staged locked deployment will unlock it.
+The deployment will then be applied automatically on the next shutdown or reboot.
+
+#### Checking for updates without side effects
+
+To check if updates are available without modifying the lock state:
+
+```shell
+bootc upgrade --check
+```
+
+This only downloads updated metadata without changing the finalization lock state.
+
+#### Example workflow
+
+A typical workflow for controlled updates:
+
+```shell
+# 1. Download and lock the update
+bootc upgrade --download-only
+
+# 2. Verify the staged deployment
+bootc status --verbose
+# Output shows: Locked: yes
+
+# 3. Test or wait for maintenance window...
+
+# 4. Apply the update (choose one):
+# Option A: Unlock and let it apply on next shutdown
+bootc upgrade
+
+# Option B: Apply immediately with reboot
+bootc upgrade --apply
+```
+
+**Important**: If you reboot before applying a locked update, the system will boot into the
```

Comment:
Also note here that if one switches to a different image, then that will replace the previous cached download-only image.